tap_appstore/streams.py

```diff
@@ -1,72 +1,150 @@
+        th.Property("Apple Identifier", th.StringType),
+        th.Property("Customer Price", th.NumberType),
+        th.Property("Promo Code", th.StringType),
+        th.Property("Parent Identifier", th.StringType),
+        th.Property("Subscription", th.StringType),
+        th.Property("Period", th.StringType),
+        th.Property("Category", th.StringType),
+        th.Property("CMB", th.StringType),
+        th.Property("Device", th.StringType),
+        th.Property("Supported Platforms", th.StringType),
+        th.Property("Proceeds Reason", th.StringType),
+        th.Property("Preserved Pricing", th.StringType),
+        th.Property("Client", th.StringType),
+        th.Property("Order Type", th.StringType)
+    ).to_dict()
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.get_records()
 
-# TODO: Delete this is if not using json files for schema definition
-SCHEMAS_DIR = importlib_resources.files(__package__) / "schemas"
-# TODO: - Override `UsersStream` and `GroupsStream` with your own stream definition.
-#       - Copy-paste as many times as needed to create multiple stream types.
+    def get_records(self, *args, **kwargs):
+        """Fetch sales reports from the App Store Connect API and save raw data."""
+        KEY_ID = self.config['key_id']
+        ISSUER_ID = self.config['issuer_id']
+        VENDOR_NUMBER = self.config['vendor']  # Updated to use the configuration
+        PATH_TO_KEY = os.path.expanduser(self.config['key_file'])
 
+        with open(PATH_TO_KEY, 'r') as f:
+            PRIVATE_KEY = f.read()
 
-class UsersStream(AppStoreStream):
-    """Define custom stream."""
+        # Create the Connection
+        connection = Connection(ISSUER_ID, KEY_ID, PRIVATE_KEY)
+        # Prepare and make the API request using the dynamically set report date
 
-    name = "users"
-    path = "/users"
-    primary_keys: t.ClassVar[list[str]] = ["id"]
-    replication_key = None
-    # Optionally, you may also use `schema_filepath` in place of `schema`:
-    # schema_filepath = SCHEMAS_DIR / "users.json"  # noqa: ERA001
-    schema = th.PropertiesList(
-        th.Property("name", th.StringType),
-        th.Property(
-            "id",
-            th.StringType,
-            description="The user's system ID",
-        ),
-        th.Property(
-            "age",
-            th.IntegerType,
-            description="The user's age in years",
-        ),
-        th.Property(
-            "email",
-            th.StringType,
-            description="The user's email address",
-        ),
-        th.Property("street", th.StringType),
-        th.Property("city", th.StringType),
-        th.Property(
-            "state",
-            th.StringType,
-            description="State name in ISO 3166-2 format",
-        ),
-        th.Property("zip", th.StringType),
-    ).to_dict()
+        try:
+            response = connection.sales_reports().filter(
```

Comment:
applaud's endpoints are pretty robust and seem easy to work with - https://github.com/codinn/applaud/blob/ed168ca67465b5c0acf4ab4f4e285a2ab348c96d/applaud/endpoints/sales_reports.py